uaclient/jobs/eol_status.py

```diff
@@ -0,0 +1,13 @@
+from uaclient import util
+from uaclient.config import UAConfig
+from uaclient.entitlements.esm import ESMInfraEntitlement
+
+
+def check_eol_and_update(cfg: UAConfig) -> bool:
+    if cfg.is_attached:
+        return True
+    series = util.get_platform_info()["series"]
+    if not util.is_active_esm(series):
+        return True
+    ESMInfraEntitlement().setup_unauthenticated_repo()
```

Comment:
@lucasmoura @renanrodrigo  Should we add ESMApps entitlement as well?

Reply:
ESM Apps will be advertised to every system, it doesn't matter if it is EOL or not. We will only add the unauthenticated repo when Apps is officially released

Reply:
Right, we don't need to call `ESMAppsEntitlement().setup_unauthenticated_repo()`